rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php

```diff
@@ -87,135 +84,70 @@ public function refactor(Node $node): ?Node
         return $this->processUnspreadOperatorMethodCallArgs($node);
     }
 
-    private function getClassFileNameByClassMethod(ClassMethod $classMethod): ?string
-    {
-        $parent = $classMethod->getAttribute(AttributeKey::PARENT_NODE);
-        if (! $parent instanceof Class_) {
-            return null;
-        }
-
-        $shortClassName = (string) $parent->name;
-        if (Strings::match($shortClassName, self::ANONYMOUS_CLASS_REGEX)) {
-            return null;
-        }
-
-        $reflectionClass = new ReflectionClass((string) $parent->namespacedName);
-        return (string) $reflectionClass->getFileName();
-    }
-
-    private function getClassFileNameByMethodCall(MethodCall $methodCall): ?string
-    {
-        $scope = $methodCall->getAttribute(AttributeKey::SCOPE);
-        if ($scope === null) {
-            return null;
-        }
-
-        $type = $scope->getType($methodCall->var);
-        if ($type instanceof ObjectType) {
-            $classReflection = $type->getClassReflection();
-            if ($classReflection === null) {
-                return null;
-            }
-
-            return (string) $classReflection->getFileName();
-        }
-
-        if ($type instanceof ThisType) {
-            $staticObjectType = $type->getStaticObjectType();
-            $classReflection = $staticObjectType->getClassReflection();
-            if ($classReflection === null) {
-                return null;
-            }
-
-            return (string) $classReflection->getFileName();
-        }
-
-        return null;
-    }
-
     private function processUnspreadOperatorClassMethodParams(ClassMethod $classMethod): ?ClassMethod
     {
-        if ($this->isInVendor($classMethod)) {
-            return null;
-        }
-
-        $params = $classMethod->params;
-        if ($params === []) {
-            return null;
-        }
-
-        $spreadVariables = $this->getSpreadVariables($params);
-        if ($spreadVariables === []) {
+        $spreadParams = $this->spreadVariablesCollector->resolveFromClassMethod($classMethod);
+        if ($spreadParams === []) {
             return null;
         }
 
-        foreach (array_keys($spreadVariables) as $key) {
-            $classMethod->params[$key]->variadic = false;
-            $classMethod->params[$key]->type = new Identifier('array');
+        foreach ($spreadParams as $spreadParam) {
+            $spreadParam->variadic = false;
+            $spreadParam->type = new Identifier('array');
         }
 
         return $classMethod;
     }
 
-    /**
-     * @param ClassMethod|MethodCall $node
-     */
-    private function isInVendor(Node $node): bool
```

Comment:
@samsonasik Just a tip for next code. The `/vendor` check can be done better by using `NodeRepository`. It contains all the nodes from parsed project. So what is not there, is in vendor :)

Reply:
Thank you